### PR TITLE
Removal of OAuth2 stuff from CLI

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
Stuff this does
 - remove CLI login subcommand
 - remove the `--conf.token` option from the CLI
 - remove applications and tokens as resources in CLI, and from import/export
   - bleeds over into the collection, as applications is duplicated in options for collection module
 - cascade delete some supporting code in the awxkit library

Ping @gravesm

##### ISSUE TYPE
 - Breaking Change 

##### COMPONENT NAME
 - CLI

##### COMPONENT NAME
 - API

### CODE REVIEW

**Comment by AlanCoding** [comment]
Highly related to: https://github.com/ansible/awx/pull/15623

**Comment by AlanCoding** [comment]
Can't get verification from integration tests due to unrelated issues.

CI failures I have addressed in other open PRs. Marked as ready for review, otherwise, merge blocked on totally unrelated stuff.

File: awx_collection/plugins/modules/export.py [hunk 1]
```diff
@@ -86,11 +86,6 @@
         - workflow names, IDs, or named URLs to export
       type: list
       elements: str
-    applications:
```
Comment (PabloHiro) [inline awx_collection/plugins/modules/export.py:89]: I did not remove the applications module from the Collection PR. I thought it was used for more purposes apart from Oauth. Is it not?

Comment (AlanCoding) [reply awx_collection/plugins/modules/export.py:89]: I didn't think applications serve any purpose other than for creating tokens. IMO, the tests seemed to reflect this. It's not impossible that I'm wrong, but it seemed relatively clear even from reading the issue for _this_ PR.

File: awxkit/awxkit/api/pages/base.py
Comment (hakbailey) [inline awxkit/awxkit/api/pages/base.py:1]: It looks like there are still some auth token related functions starting on line 135 and a class on line 13, should we remove those?

**Review by hakbailey — COMMENTED** [review]
Just double checking to be thorough, should we remove any of the following?

- `Token_Auth` class in [/awxkit/awxkit/api/client.py](https://github.com/ansible/awx/blob/65d211244bb128b76734de33b3c37fdc26e3e9df/awxkit/awxkit/api/client.py#L16)
- Reference to `OAuth2AccessToken` in [/awxkit/awxkit/awx/utils.py](https://github.com/ansible/awx/blob/65d211244bb128b76734de33b3c37fdc26e3e9df/awxkit/awxkit/awx/utils.py#L86)
- Reference to `load_authtoken()` function in [/awxkit/awxkit/scripts/basic_session.py](https://github.com/ansible/awx/blob/65d211244bb128b76734de33b3c37fdc26e3e9df/awxkit/awxkit/scripts/basic_session.py#L66)
- Usage of token in [/awxkit/awxkit/ws.py](https://github.com/ansible/awx/blob/65d211244bb128b76734de33b3c37fdc26e3e9df/awxkit/awxkit/ws.py#L89)

**Comment by AlanCoding** [comment]
@hakbailey thanks, these are good suggestions. However, I have been trying to follow the de-referencing cascade from removing those, which goes deep into the fixture structure of integration tests. This gets a little crazy. My head is spinning from the distinction of `v2.load_session()` and `v2.connection.login()`, and if the distinction between those should remain after removing tokens.

I deleted the method in awxkit

```python
def uses_sessions(connection):
    session_login = connection.get(f"{config.api_base_path}login/")
    return session_login.status_code == 200
```

But this is really my critical point of uncertainty. Why would, or would not, the login endpoint give a 200 code? I thought this would have been part of the recent resource-server work, but no, the method appears to be positively ancient. Previously, it should have _always_ given a 200.

I'm thinking perhaps this goes too far, since `AWXKIT_SESSIONS` is its own independent configuration. You could use a session, use basic auth, or use a token. The session/not-session is still valid? I think...

**Comment by AlanCoding** [comment]
@hakbailey I pushed a commit that carries out your suggestions. This isn't super-trivial, because there were other method references beyond _just_ the things you mention, and there are many corresponding integration test changes that also cascade from those.

**Review by hakbailey — APPROVED** [review]
This looks good to me assuming the integration test updates worked...

File: awxkit/awxkit/scripts/basic_session.py [hunk 2]
```diff
@@ -59,11 +59,8 @@ def main():
 
         global root
         root = api.Api()
-        if uses_sessions(root.connection):
-            config.use_sessions = True
-            root.load_session().get()
-        else:
-            root.load_authtoken().get()
+        config.use_sessions = True
+        root.load_session().get()
```
Comment (AlanCoding) [inline awxkit/awxkit/scripts/basic_session.py]: I'm going to revert this change, because it is the mostly likely candidate for the stuff that has been going wrong in testing. Using sessions vs not is still a valid thing, and this is probably mistaken.

**Comment by AlanCoding** [comment]
Moved to feature branch
